src/git/Buffer.cpp

```diff
@@ -8,13 +8,19 @@
 //
 
 #include "Buffer.h"
+#include <QByteArray>
 
 namespace git {
 
 Buffer::Buffer(const char *data, int size)
   : d(GIT_BUF_INIT_CONST(data, size))
 {}
 
+
+Buffer::Buffer(const QByteArray &data)
+  : d(GIT_BUF_INIT_CONST(data.constData(), data.size()))
```

Comment:
Might be cleaner to call the other constructor instead (don't have to repeat `GIT_BUF_INIT_CONST` and whatever people insert in the `{}` in the future).

Reply:
It was a convenience, can remove.

Reply:
Removing is fine. FTR, what I meant was calling the first constructor from the second one.